internal/config/parser/replacer.go

```diff
@@ -0,0 +1,39 @@
+package parser
+
+import (
+	"strings"
+)
+
+var knownMap map[string]string
+var unknownMap map[string]bool
+
+func init() {
+	knownMap = make(map[string]string)
+	unknownMap = make(map[string]bool)
+	for _, v := range known {
+		knownMap[v[0]] = v[1]
+	}
+}
+
+type camelCaseReplacer struct{}
+
+func (_ camelCaseReplacer) Replace(str string) string {
+	// convert to known variable name
+	if k, ok := knownMap[str]; ok {
+		if k == "" {
+			return strings.ToUpper(strings.ReplaceAll(str, ".", "_"))
+		}
+		return k
+	}
+	// otherwise replace automatically but warn the user
+	unknownMap[str] = true
+	return strings.ToUpper(strings.ReplaceAll(str, ".", "_"))
+}
+
+func UnknownEnvVariables() []string {
+	result := make([]string, 0, len(unknownMap))
+	for k := range unknownMap {
+		result = append(result, k)
+	}
+	return result
+}
```

Comment:
Yeah, it's horrible. Not my fault, this is how viper interface makes it hard to use it. Alternatively, let's remove viper and just use `os.GetEnv` and few functions that will convert from string to our config structure with types.

Reply:
I am all for it, with all these additions viper kinda looses its appeal of minimal configuration code. Lets get rid of it and do it ourselves?